### PR TITLE
Navigates to podcast view when tapping on podcast name on Player.

### LINKED ISSUE #1322: Make "Podcast title" tappable in the player and link to podcast screen
### Discussed in https://github.com/Automattic/pocket-casts-ios/discussions/1313

<div type='discussions-op-text'>

<sup>Originally posted by **david-gonzalez-a8c** December 22, 2023</sup>
Let's make the "Podcast title" in the player tappable and take to the podcast view. This is something that most products in the space have and it makes sense:

image

I don't think we need to make further adjustments in the spacing to make it possible.</div>

### PR DESCRIPTION
**Pull request by SergioEstevao**
Fixes #1322

This PR adds the possibility to navigate to the Podcast page when tapping on the podcast name on the Player.

https://github.com/Automattic/pocket-casts-ios/assets/651601/69f3f19d-35bb-40e3-8e43-f774da0461ff

## To test

 - Start the app
 - Start listening to a podcast episode without chapters
 - Open the player full screen
 - Tap on the podcast title
 - Check that the view for podcast opens

## Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to `CHANGELOG.md` if necessary.
- [x] I have considered adding unit tests for my changes.
- [x] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.

### CODE REVIEW

**Comment by pocketcasts** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">PR requires at least one label.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          3 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot is included in the pull request. Consider adding some for clarity.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">No reviewers have been set for this PR yet. Please request a review from <strong>@ Automattic/pocket-casts-ios</strong>.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by pocketcasts** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">PR requires at least one label.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          3 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot is included in the pull request. Consider adding some for clarity.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">No reviewers have been set for this PR yet. Please request a review from <strong>@ Automattic/pocket-casts-ios</strong>.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: podcasts/NowPlayingPlayerItemViewController.swift [hunk 1]
```diff
@@ -289,6 +294,10 @@ class NowPlayingPlayerItemViewController: PlayerItemViewController {
         containerDelegate?.scrollToCurrentChapter()
     }
 
+    @objc private func podcastNameTapped() {
+        containerDelegate?.navigateToPodcast()
```
Comment (SergioEstevao) [inline podcasts/NowPlayingPlayerItemViewController.swift:299]: Should I add a Analytics event tracking on this point?

**Comment by pocketcasts** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot is included in the pull request. Consider adding some for clarity.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: podcasts/Analytics/AnalyticsEvent.swift [hunk 2]
```diff
@@ -335,6 +335,7 @@ enum AnalyticsEvent: String {
     case playerTabSelected
     case playerShowNotesLinkTapped
     case playerChapterSelected
+    case playerPodcastLinkTapped
```
Comment (bjtitus) [inline podcasts/Analytics/AnalyticsEvent.swift]: I wonder if this should be `PodcastNameTapped` to mirror the `episode_detail_podcast_name_tapped` event in the Episode Detail page when tapping the name.

**Review by bjtitus — APPROVED** [review]
Code looks good and everything works great! This has been one of my biggest annoyances using Pocket Casts, honestly. 😆 

I have two pieces of minor, optional, feedback:

I'm a bit inexperienced in the event names, but I feel like the name should be `podcast_name_tapped` to match `episode_detail_podcast_name_tapped`. I think the other `linkTapped` event is for actual hyperlinks in the show notes.

I wonder if there should be a chevron next to the title to make it obvious that it's tappable like there is on the Episode Detail page:

5CD397F5-7C48-4F47-A15C-454FA5055657_1_201_a

**Comment by bjtitus** [comment]
Would it make sense to pass through `animated` to the `navigateTo` methods? It looks a bit better to me to animate the miniplayer dismissal but that may impact navigation other than this one spot:


https://github.com/Automattic/pocket-casts-ios/assets/3250/7c667d7f-0a87-4fb5-86ab-ea3f7aeb5f19

Here's the quick hack of a patch for just this one navigation spot:
[animated_navigateto_podcast.patch](https://github.com/Automattic/pocket-casts-ios/files/13879915/animated_navigateto_podcast.patch)

**Comment by SergioEstevao** [comment]
> I wonder if there should be a chevron next to the title to make it obvious that it's tappable like there is on the Episode Detail page:

@leandroalonso what do you think about this ^? 

From what I see on chapters, they are also tappable and we don't show a chevron there.

**Comment by leandroalonso** [comment]
@SergioEstevao this is a question for @david-gonzalez-a8c!

**Comment by david-gonzalez-a8c** [comment]
No chevron, I feel like this is standard enough and I prefer to keep this screen cleaner.

File: podcasts/Analytics/AnalyticsEvent.swift (hunk 2 above)
Comment (SergioEstevao) [reply podcasts/Analytics/AnalyticsEvent.swift]: I will change it.

**Comment by pocketcasts** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot is included in the pull request. Consider adding some for clarity.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by pocketcasts** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot is included in the pull request. Consider adding some for clarity.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Pull request merged**